assets/src/edit-story/components/inspector/document/publish/index.js

```diff
@@ -112,12 +132,39 @@ function PublishPanel() {
     [updateStory]
   );
 
-  const handleChangeValue = useCallback(
-    (prop) => (value) => updateStory({ properties: { [prop]: value } }),
+  useEffect(() => {
+    if (users?.length) {
+      const currentAuthor = users.find(({ id }) => author === id);
+      if (!currentAuthor) {
+        getUserById(author)
+          .then(({ id, name }) => {
+            setVisibleOptions([{ id, name }, ...users]);
+          })
+          .catch(() => setVisibleOptions(users));
```

Comment:
Don't we have the full author information from the story object already? If not, we should fix that. (i.e. passing `?_embed=author` when loading the story in `useLoadStory` and in `edit-story.php`)

That avoids having to do an actual HTTTP request here and calling `setVisibleOptions`

Reply:
Happy to push some code to better explain my thoughts here if you'd like

Reply:
I actually started doing with the `_embed` version at first, however, it's possible that the author changes to something without having the full information, and then we'd still need to request the new information. I figured that since it's not something that's used very often then requesting this is perhaps not that big of an influence.

The reason for adding this extra request was mainly for when the author is changed, the original we could indeed get from embedding author. Thoughts?

Reply:
How/when would the author change to something without having the full information?

Sounds like we should call

```
updateStory({
  properties: { author: id },
});
```

with more than just the ID?

Reply:
> How/when would the author change to something without having the full information?

For example, when we have 200 users, and we're changing the author to be nr 200 in the list, then additional information about this author will not be available just by the ID (since it's not among the first 100 loaded authors), and also not in the embed, since the DB is not updated yet.

We could indeed update the story by passing forward both ID and the name from the dropdown option upon selection (where it is available!) instead of just the ID as it is now, however, then we would need to generally store the author as an object and not just an ID. Is that what you were thinking of -- storing the `author` as an object for the story instead of just the ID?

I suppose we could pass just the ID when saving the story to DB, and convert it to an object when it's received from the DB, getting the `name` from the embedded author.

Happy to apply this if you think it's a better option.

Reply:
> Is that what you were thinking of -- storing the author as an object for the story instead of just the ID?

yep!

> I suppose we could pass just the ID when saving the story to DB, and convert it to an object when it's received from the DB, getting the name from the embedded author.

I think this makes sense